keanu-python/keanu/sequence.py

```diff
@@ -59,17 +62,30 @@ def __init__(self,
                 "Cannot create a sequence of an unknown size: you must specify either a count of a data_generator")
         elif count is not None and data_generator is not None:
             raise ValueError("If you pass in a data_generator you cannot also pass in a count")
+        elif factory is None and factories is None:
+            raise ValueError("You must provide a value for either the 'factory' or 'factories' input")
 
         if count is not None:
-            function = lambda p: factory(SequenceItem(p))
-            consumer = Consumer(function)
-            builder = builder.count(count).withFactory(consumer)
+            if factories is not None:
+                functions = [Consumer(lambda p, f=f: f(SequenceItem(p))) for f in factories]
+                java_functions = ListConverter().convert(functions, k._gateway._gateway_client)
+                builder = builder.count(count).withFactories(java_functions)
+            elif factory is not None:
```

Comment:
Having 2 code paths here (and on line 84) increases the complexity. Instead you can just turn a singleton `factory` into a list at the start of this function:

    if factory is not None:
        factories = [factory]

Reply:
Or if you implement my suggestion to allow `factories` to be either a `Consumer` or an `Iterable[Consumer]` then you'd do:

    if isinstance(factories, Consumer):
        factories = [factories]